### ISSUE TITLE
Cannot read properties of undefined (reading 'build')

### ISSUE THREAD

**Issue by gregorwolf** [mikezaschka/cds-dbm#294]
Hi @mikezaschka and @ryegrosT8,

I've created the branch [feat/update-to-cds-6](https://github.com/gregorwolf/pg-beershop/tree/feat/update-to-cds-6) for my pg-beershop project to upgrade to the latest CAP, cds-pg and cds-dbm version. But when I try to run `npm run build:cf` in my project I get the output:

```
[INTERNAL ERROR] TypeError: Cannot read properties of undefined (reading 'build')     
at PostgresCfModuleBuilder.<anonymous> (node_modules/cds-dbm/dist/build/postgres-cf/index.js:81:97)
```

@ryegrosT8 did you have a working build and deploy to BTP CF?

Best Regards
Gregor

**Comment by ryegrosT8** [comment]
Hi @gregorwolf ,

Actually, I didn't deploy my project yet to BTP after the last changes.

But I think I got what is happening, the PostgresCfModuleBuilder extends BuildTaskHandlerInternal of @sap/cds package, in the previous version there was a public property called "env" that was "wrapping” the "cds.env". 

I checkout your https://github.com/gregorwolf/pg-beershop/tree/feat/update-to-cds-6 and made a change directly at node_modules/cds-dbm/dist/build/postgres-cf/index.js and seems to work.

Would you mind helping me to test if there is anything else wrong?

Could you please replace the line 81 at  node_modules/cds-dbm/dist/build/postgres-cf/index.js from:
```
            return ((extname !== FILE_EXT_CSV && extname !== FILE_EXT_CDS && entry !== this.env.build.outputfile) ||
```
to:
```
            return ((extname !== FILE_EXT_CSV && extname !== FILE_EXT_CDS && entry !== cds_1.default.build['outputfile']) ||
```
And see if there is anything else out of expected by running pgbeershop?

I will submit a new PR here tomorrow if it works well for you.

**Comment by gregorwolf** [comment]
Hi @ryegrosT8,

yes, this fixes the build issue. I still had the issue that the deployed srv did not start anymore with "cds run" so I changed that to ""node node_modules/@sap/cds/bin/cds.js run".

CU
Gregor

**Comment by gregorwolf** [comment]
The deployment in the db module is failing with:

`Error: [cds-dbm] - failed to load model undefined`

I guess it is the same root cause because in the deployed version the patch has not been applied. Please check my PR #295.

**Comment by ryegrosT8** [comment]
Hi @gregorwolf , 

Sorry to be late, I got a busy day here today and wasn’t able to submit the PR. 

I guess that should be "cds.env.build['output file']" instead of "cds.default.build['output file']", isn't right?

**Comment by gregorwolf** [comment]
Please check my update @ryegrosT8. Maybe you have some time to check if it's working.

**Comment by ryegrosT8** [comment]
Still with error when try to run ```npm run build:cf```, the last commit https://github.com/mikezaschka/cds-dbm/pull/295/commits/ccca821027383d67c9462599c2342bc890c01610 seems to undo the rigth one https://github.com/mikezaschka/cds-dbm/pull/295/commits/6e26f639b601f4d3c4ddd50b38f6563b24198d95. 

About the error ```Error: [cds-dbm] - failed to load model undefined```` , I had a similar error in my project, could you try to especify the folders that holds your cds database files here at your package.json and check if it works?

![image](https://user-images.githubusercontent.com/120337840/219208661-fc1ac8e9-42a2-406e-9d4e-66b60cdae221.png)

**Comment by gregorwolf** [comment]
@ajewelbd would be great if you can provide the correction in the TypeScript source.

**Comment by mikezaschka** [comment]
Fixed in version 1.11.
Thank you @gregorwolf, @ajewelbd and @ryegrosT8.
